src/tests/roc_library/test_sender_receiver.cpp

```diff
@@ -11,534 +11,282 @@
-        sender_conf.resampler_profile = ROC_RESAMPLER_DISABLE;
-        sender_conf.packet_length =
-            PacketSamples * 1000000000ul / (SampleRate * NumChans);
-        if (flags & FlagFEC) {
-            sender_conf.fec_code = ROC_FEC_RS8M;
-            sender_conf.fec_block_source_packets = SourcePackets;
-            sender_conf.fec_block_repair_packets = RepairPackets;
-        } else {
-            sender_conf.fec_code = ROC_FEC_DISABLE;
-        }
+    test::Receiver receiver_1(context, receiver_conf, sample_step, test::FrameSamples,
+                              Flags);
 
-        memset(&receiver_conf, 0, sizeof(receiver_conf));
-        receiver_conf.frame_sample_rate = SampleRate;
-        receiver_conf.frame_channels = ROC_CHANNEL_SET_STEREO;
-        receiver_conf.frame_encoding = ROC_FRAME_ENCODING_PCM_FLOAT;
-        receiver_conf.clock_source = ROC_CLOCK_INTERNAL;
-        receiver_conf.resampler_profile = ROC_RESAMPLER_DISABLE;
-        receiver_conf.target_latency = Latency * 1000000000ul / SampleRate;
-        receiver_conf.no_playback_timeout = Timeout * 1000000000ul / SampleRate;
-    }
+    test::Sender sender_1(context, sender_conf, receiver_1.source_endpoint(),
+                          receiver_1.repair_endpoint(), sample_step, test::FrameSamples,
+                          Flags);
 
-    bool is_fec_supported() {
-        return fec::CodecMap::instance().is_supported(packet::FEC_ReedSolomon_M8);
-    }
-};
+    sender_1.start();
+    receiver_1.run();
+    sender_1.stop();
+    sender_1.join();
 
-TEST(sender_receiver, bare_rtp) {
-    enum { Flags = 0 };
+    test::Receiver receiver_2(context, receiver_conf, sample_step, test::FrameSamples,
+                              Flags);
 
-    init_config(Flags);
+    test::Sender sender_2(context, sender_conf, receiver_2.source_endpoint(),
+                          receiver_2.repair_endpoint(), sample_step, test::FrameSamples,
+                          Flags);
 
-    Context context;
+    sender_2.start();
+    receiver_2.run();
+    sender_2.stop();
+    sender_2.join();
 
-    Receiver receiver(context, receiver_conf, sample_step, FrameSamples, Flags);
+    test::Receiver receiver_3(context, receiver_conf, sample_step, test::FrameSamples,
+                              Flags);
 
-    Sender sender(context, sender_conf, receiver.source_endpoint(),
-                  receiver.repair_endpoint(), sample_step, FrameSamples, Flags);
+    test::Sender sender_3(context, sender_conf, receiver_3.source_endpoint(),
+                          receiver_3.repair_endpoint(), sample_step, test::FrameSamples,
+                          Flags);
 
-    sender.start();
-    receiver.run();
-    sender.stop();
-    sender.join();
+    sender_3.start();
+    receiver_3.run();
+    sender_3.stop();
+    sender_3.join();
 }
 
-TEST(sender_receiver, fec_without_losses) {
-    if (!is_fec_supported()) {
+TEST(sender_receiver, ldpc_without_losses) {
```

Comment:
I suggest to group reed-solomon and LDPC tests together, e.g. move `ldpc_without_losses` and `ldpc_with_losses` right after `fec_with_losses` (which should be renamed to `rs8m_with_losses I guess).